File: 프로그래머스/0/181838. 날짜 비교하기/날짜 비교하기.js
function solution(date1, date2) {
    // 연도 비교
    if (date1[0] < date2[0]) {
        return 1;
    } else if (date1[0] > date2[0]) {
        return 0;
    }

    // 월 비교
    if (date1[1] < date2[1]) {
        return 1;
    } else if (date1[1] > date2[1]) {
        return 0;
    }

    // 날짜 비교
    if (date1[2] < date2[2]) {
        return 1;
    } else {
        return 0;
    }
}

File: 프로그래머스/0/181838. 날짜 비교하기/날짜 비교하기.py
def solution(date1, date2):
    
    y1, m1, d1 = date1
    y2, m2, d2 = date2
    
    if y1 != y2:
        return 1 if y1 < y2 else 0
    if m1 != m2:
        return 1 if m1 < m2 else 0
    return 1 if d1 < d2 else 0

# return 1 if date1 < date2 else 0